### PR TITLE
ci: add stale bot

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Long Dai <long0dai@foxmail.com>

# Description

Add stale bot.

## Issue reference

Part of #3245

### CODE REVIEW

**Review by artursouza — CHANGES_REQUESTED** [review]
Thanks for this. Please, combine this with the automerge workflow - it already runs on a schedule.

**Review by tanvigour — COMMENTED** [review]
Thanks a lot for this. As Artur mentioned above, please combine it with auto merge.

**Comment by daixiang0** [comment]
@artursouza combine them and rename a meaningful name.

File: .github/workflows/stale.yml
Comment (CodeMonkeyLeet) [inline .github/workflows/stale.yml:1]: This file should probably be removed given the combined dapr-schedule.yml?

File: .github/workflows/stale.yml
Comment (CodeMonkeyLeet) [inline .github/workflows/stale.yml:22]: Suggestion: Several projects used the `pinned` label for this purpose

@artursouza Whatever the name, a maintainer needs to add the appropriate label for the documentation to be accurate.

File: .github/workflows/stale.yml
Comment (daixiang0) [reply .github/workflows/stale.yml:1]: Sure.

File: .github/workflows/stale.yml
Comment (daixiang0) [reply .github/workflows/stale.yml:22]: It maybe need to discuss which label we use.

Comment (artursouza) [reply .github/workflows/stale.yml:22]: Let's use the `pinned` label.

Comment (artursouza) [reply .github/workflows/stale.yml:22]: When this runs for the first time, will it close all issues and PRs right away or will it give the 7 days notice first? I want to know if we want to label the existing PRs and issues now or later.

Comment (artursouza) [reply .github/workflows/stale.yml:22]: I am also adding the `triage/resolved` label as exception.

Comment (daixiang0) [reply .github/workflows/stale.yml:22]: @artursouza should add label now.

Comment (artursouza) [reply .github/workflows/stale.yml:22]: Sorry, I just realized there are two workflows. Do we still need stale.yml? I thought we were merging all in one.

Comment (daixiang0) [reply .github/workflows/stale.yml:22]: Sure, I will merge them.

**Comment by daixiang0** [comment]
@artursouza could it merge?

**Comment by artursouza** [comment]
Not yet. We need to visit every PR and issue first and label them to avoid a mass closing of issues and PRs. This is actually a pretty big task :)

**Comment by daixiang0** [comment]
There is a big time gap between add stale label and close, we can check one by one rather than update all pr.

**Comment by artursouza** [comment]
> There is a big time gap between add stale label and close, we can check one by one rather than update all pr.

Even for items that are older than 30 days, does stale bot only close after the first notification?

**Comment by daixiang0** [comment]
It would add a stale label, and wait for the specified time, it close, so we have long time to mark them after merged.

**Review by artursouza — APPROVED** [review]

**Review by artursouza — APPROVED** [review]

**Comment by daixiang0** [comment]
@artursouza sorry for force push :(

**Review by artursouza — APPROVED** [review]

**Pull request merged**
